developing/bigbrother/start_pyro4bot_BB.py

```diff
@@ -178,13 +178,13 @@ def request(self, obj, claimant):
                 self.claimant_list.append(claimant)
 
                 if (DEBUGGER):
-                    print(colored("Components waiting:", "green"))
+                    print((colored("Components waiting:", "green")))
                     pprint.pprint(ast.literal_eval(
                         json.dumps(self.claimant_list)))
-                    print(colored("Components that are needed:", "green"))
-                    for x in self.async_waitings.keys():
-                        print("{}. Type: {}. Needed by: {}".format(
-                            x, self.async_waitings[x]["target_type"], self.async_waitings[x]["claimant"]))
+                    print((colored("Components that are needed:", "green")))
+                    for x in list(self.async_waitings.keys()):
```

Comment:
You are unnecessarily creating an entire list, when using the iterator `keys()` is enough. You can also omit `.keys()`, since by default, iterating over a `dict` yields keys, but it's good to be explicit.